pkg/provider-local/machine-provider/local/driver.go

```diff
@@ -55,15 +55,20 @@ func podForMachine(machine *machinev1alpha1.Machine) *corev1.Pod {
 	}
 }
 
-func userDataSecretForMachine(machine *machinev1alpha1.Machine) *corev1.Secret {
+func userDataSecretForMachine(machine *machinev1alpha1.Machine, machineClass *machinev1alpha1.MachineClass) *corev1.Secret {
+	namespace := machine.Namespace
+	// machine.Namespace may be empty due to machine controller manager omitting namespace
```

Comment:
Do you want to link https://github.com/gardener/machine-controller-manager/pull/932 here?

Reply:
Can we also add a TODO to revert this PR after we adopt https://github.com/gardener/machine-controller-manager/pull/932?

Reply:
Added `TODO` and link to the machine controller manager PR in https://github.com/gardener/gardener/pull/10176/commits/feadd3fbbc16d3e2a72dea53243a45fbdec40caa.